tests/unit/tracing_test.cpp

```diff
@@ -0,0 +1,74 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+#include "mocking.h"
+
+using namespace testing;
+
+extern "C" {
+   #include <tilck/kernel/test/tracing.h>
+}
+
+extern bool *traced_syscalls;
```

Comment:
The whole point of introducing `include/tilck/kernel/test/tracing.h` and all the files there is to avoid unchecked declartions like those in the unit tests. Move the `extern bool *traced_syscalls;` to that file as it is. While for `traced_syscalls_str` you need to declare it there with STATIC.

Reply:
Fixed.